### PR TITLE
Vale linter config

### PR DESCRIPTION
**Pull request by jstirnaman**
Initial commit of Vale linter config for spelling and grammar. This is still WIP, but helpful enough to start using and refining in local environments.

- [ ] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/docs-v2/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [ ] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/cloud-dedicated/.vale.ini [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+Vocab = InfluxData, Cloud-Dedicated
+
+MinAlertLevel = warning
+
+Packages = Google, Hugo
+
+[*.md]
+BasedOnStyles = InfluxDataDocs, Google
+
+Google.Headings.Capitalization = NO
```
Comment (jstirnaman) [inline content/influxdb/cloud-dedicated/.vale.ini:12]:
```suggestion
Google.Headings = NO
```

File: content/influxdb/cloud-serverless/.vale.ini [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Vocab = InfluxData, Cloud-Serverless
+
+MinAlertLevel = warning
+
+Packages = Google, Hugo
+
+[*.md]
+BasedOnStyles = InfluxDataDocs, Google
+
+Google.Headings.Capitalization = NO
```
Comment (jstirnaman) [inline content/influxdb/cloud-serverless/.vale.ini:12]:
```suggestion
Google.Headings = NO
```

File: .ci/vale/styles/Vocab/Cloud-Serverless/reject.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Cloud Dedicated
+cloud-dedicated
+database name
```
Comment (sanderson) [inline .ci/vale/styles/Vocab/Cloud-Serverless/reject.txt:3]:
```suggestion
We need to decide how to handle this with DBRP mapping and people coming from 1.x to serverless.
```

**Review by sanderson — CHANGES_REQUESTED** [review]
Very cool. Some questions and suggestions. How deep in the hole are we after running the linter? 😄

File: .ci/vale/styles/Google/WordList.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  approx\.: approximately
+  authN: authentication
+  authZ: authorization
+  autoupdate: automatically update
+  cellular data: mobile data
+  cellular network: mobile network
+  chapter: documents|pages|sections
+  check box: checkbox
+  check: select
+  CLI: command-line tool
```
[the remark this answers is not in the document]
Comment (jstirnaman) [reply .ci/vale/styles/Google/WordList.yml:42]: > I think this is going to get really annoying 😄. I don't know that I want to replace every instance of "CLI" with "command-line tool."

I thought exactly the same thing

File: .ci/vale/styles/Google/WordList.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  click on: click|click in
+  Cloud: Google Cloud Platform|GCP
```
[the remark this answers is not in the document]
Comment (jstirnaman) [reply .ci/vale/styles/Google/WordList.yml:44]: > Vendor-specific. We need to remove this one.
Yeah, that def generates a lot of noise.

File: .ci/vale/styles/Google/DateFormat.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+extends: existence
+message: "Use 'July 31, 2016' format, not '%s'."
+link: 'https://developers.google.com/style/dates-times'
+ignorecase: true
+level: error
+nonword: true
+tokens:
```
[the remark this answers is not in the document]
Comment (jstirnaman) [reply .ci/vale/styles/Google/DateFormat.yml:7]: We can disable the rule, though honestly I didn't see it throw. Appears to check for non-U.S. date formats (`12-Jan-2023` or `12-01-2023`)

File: .ci/vale/styles/Google/Headings.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
+extends: capitalization
+message: "'%s' should use sentence-style capitalization."
+link: 'https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings'
+level: warning
+scope: heading
+match: $sentence
+indicators:
+  - ':'
+exceptions:
```
[the remark this answers is not in the document]
Comment (jstirnaman) [reply .ci/vale/styles/Google/Headings.yml:9]: Yeah or create our own to use additionally

File: .ci/vale/styles/Google/WordList.yml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  firewalls: firewall rules
+  functionality: capability|feature
+  Google account: Google Account
+  Google accounts: Google Accounts
+  Googling: search with Google
+  grayed-out: unavailable
+  HTTPs: HTTPS
+  in order to: to
+  ingest: import|load
+  k8s: Kubernetes
```
[the remark this answers is not in the document]
Comment (jstirnaman) [reply .ci/vale/styles/Google/WordList.yml:63]: The Vocab `accept.txt` files are preferred words that override these, IIUC.

File: .ci/vale/styles/Google/DateFormat.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (sanderson) [reply .ci/vale/styles/Google/DateFormat.yml:7]: Cool. Just wanted to double check.

**Review by sanderson — APPROVED** [review]

**Pull request merged**